gcc/rust/rust-session-manager.cc

```diff
@@ -474,6 +498,26 @@ Session::enable_dump (std::string arg)
 void
 Session::parse_files (int num_files, const char **files)
 {
+  rust_assert (num_files > 0);
+
+  if (options.crate_name.empty ())
+    {
+      /* HACK: We use the first file to infer the crate name, which might be
+       * incorrect: since rustc only allows one file to be supplied in the
+       * command-line */
```

Comment:
I actually would like to add the following changes so that this assumption will always be true:

```diff
diff --git a/gcc/rust/rust-lang.cc b/gcc/rust/rust-lang.cc
index 5ecd79b15..5f06862eb 100644
--- a/gcc/rust/rust-lang.cc
+++ b/gcc/rust/rust-lang.cc
@@ -168,6 +168,12 @@ grs_langhook_parse_file (void)
 {
   rust_debug ("Preparing to parse files. ");
 
+  /* Rust only allows the "root file" to be supplied */
+  if (num_in_fnames > 1) {
+    rust_error_at(Location(), "multiple input filenames provided");
+    return;
+  }
+
   Rust::Session::get_instance ().parse_files (num_in_fnames, in_fnames);
 }
 
```

Reply:
I think this should probably be done in a separate PR to help discussion and avoid cluttering. I personally think it would be a good idea